invenio_vocabularies/contrib/awards/schema.py

```diff
@@ -9,26 +9,53 @@
 """Awards schema."""
 
 from flask_babelex import lazy_gettext as _
+from functools import partial
 from marshmallow import Schema, ValidationError, fields, validate, \
     validates_schema
-from marshmallow_utils.fields import SanitizedUnicode
+from marshmallow_utils.fields import IdentifierSet, SanitizedUnicode
+from marshmallow_utils.schemas import IdentifierSchema
 
+from .config import award_schemes
 from ..funders.schema import FunderRelationSchema
+from ...services.schema import BaseVocabularySchema, i18n_strings
 
 
-class AwardRelationSchema(Schema):
+class AwardSchema(BaseVocabularySchema):
     """Award schema."""
-
-    title = SanitizedUnicode(
-        required=True,
-        validate=validate.Length(min=1, error=_('Title cannot be blank.'))
-    )
+    identifiers = IdentifierSet(fields.Nested(
+        partial(
+            IdentifierSchema,
+            allowed_schemes=award_schemes,
+            identifier_required=False
+        )
+    ))
     number = SanitizedUnicode(
         required=True,
         validate=validate.Length(min=1, error=_('Number cannot be blank.'))
     )
-    scheme = SanitizedUnicode()
-    identifier = SanitizedUnicode()
+    funder = fields.Nested(FunderRelationSchema)
+
+
+class AwardRelationSchema(Schema):
+    """Award relation schema."""
+
+    id = SanitizedUnicode()
+    number = SanitizedUnicode()
+    title = i18n_strings
+
+    @validates_schema
+    def validate_data(self, data, **kwargs):
+        """Validate either id or number/title are present."""
+        id_ = data.get("id")
+        number = data.get("number")
+        title = data.get("title")
+        if (id_) or (number and title):
```

Comment:
This is more than okay, just some options... this reaaaally depends on your coding style (and there is no correct answer here).
```suggestion
        if id_ or (number and title):
```

This would also work, and since the values are used **only** for a condition:
```python
is_id = data.get("id")
is_custom = data.get("number") and data.get("title")
if not is_id and not is_custom:
    raise
```

In any case, I would invert the conditions to avoid the `pass + else` and just have one `if`

Reply:
Ended up with something similar, without changing the variables instantiation:

```python
 if not id_ and not (number and title):
    # raise exception      
```